oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/Event.java

```diff
@@ -153,8 +154,15 @@ public boolean combine(final Metrics metrics) {
      * @since 9.0.0 Limit the length of {@link #parameters}
      */
     public void setParameters(String parameters) {
-        this.parameters = parameters == null || parameters.length() <= PARAMETER_MAX_LENGTH ?
-            parameters : parameters.substring(0, PARAMETER_MAX_LENGTH);
+        if (parameters == null || parameters.length() <= PARAMETER_MAX_LENGTH) {
+            this.parameters = parameters;
+        } else {
+            try {
+                this.parameters = StringUtil.trimJson(parameters, PARAMETER_MAX_LENGTH);
+            } catch (Exception e) {
```

Comment:
I can't see the point of this catch. What do you want to do about this case? When is the exception thrown?

Reply:
I am not quite sure that the original parameters must be in json format

Reply:
@kezhenxu94 Is the JSON format required by the protocol?

Reply:
If not, `JsonParser.parseString(jsonString)` may throw `JsonSyntaxException`